### PR TITLE
Update to sdk-java RC1

### PR DESCRIPTION
**Pull request by pierDipi**

### CODE REVIEW

**Comment by pierDipi** [comment]
```
SEVERE: Unhandled exception
io.cloudevents.rw.CloudEventRWException: Invalid specversion: 9000.1
	at io.cloudevents.rw.CloudEventRWException.newInvalidSpecVersion(CloudEventRWException.java:87)
	at io.cloudevents.SpecVersion.parse(SpecVersion.java:76)
	at io.cloudevents.core.message.impl.MessageUtils.parseStructuredOrBinaryMessage(MessageUtils.java:57)
	at io.cloudevents.http.vertx.VertxMessageFactory.createReader(VertxMessageFactory.java:42)
	at io.cloudevents.http.vertx.VertxMessageFactory.lambda$createReader$4(VertxMessageFactory.java:63)
	at io.vertx.core.impl.future.FutureImpl$1.onSuccess(FutureImpl.java:91)
	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:53)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:83)
	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:199)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:50)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:180)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.http.impl.HttpEventHandler.handleEnd(HttpEventHandler.java:79)
	at io.vertx.core.http.impl.Http1xServerRequest.onEnd(Http1xServerRequest.java:549)
	at io.vertx.core.http.impl.Http1xServerRequest.handleEnd(Http1xServerRequest.java:535)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:73)
	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:189)
	at io.vertx.core.http.impl.Http1xServerConnection.onEnd(Http1xServerConnection.java:200)
	at io.vertx.core.http.impl.Http1xServerConnection.onContent(Http1xServerConnection.java:187)
	at io.vertx.core.http.impl.Http1xServerConnection.handleOther(Http1xServerConnection.java:156)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:144)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:151)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:144)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.websocketx.extensions.WebSocketServerExtensionHandler.channelRead(WebSocketServerExtensionHandler.java:101)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by pierDipi** [comment]
```
11:58:37.648 [vert.x-eventloop-thread-1] ERROR io.vertx.core.impl.ContextImpl - Unhandled exception
io.cloudevents.rw.CloudEventRWException: Could not parse. Unknown encoding. Invalid content type or spec version
	at io.cloudevents.rw.CloudEventRWException.newUnknownEncodingException(CloudEventRWException.java:155)
	at io.cloudevents.core.message.impl.MessageUtils.parseStructuredOrBinaryMessage(MessageUtils.java:61)
	at io.cloudevents.http.vertx.VertxMessageFactory.createReader(VertxMessageFactory.java:42)
	at io.cloudevents.http.vertx.VertxMessageFactory.lambda$createReader$4(VertxMessageFactory.java:69)
	at io.vertx.core.impl.future.FutureImpl$1.onSuccess(FutureImpl.java:91)
	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:53)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:83)
	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:199)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:50)
	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:180)
	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
	at io.vertx.core.http.impl.HttpEventHandler.handleEnd(HttpEventHandler.java:79)
	at io.vertx.core.http.impl.Http1xServerRequest.onEnd(Http1xServerRequest.java:549)
	at io.vertx.core.http.impl.Http1xServerRequest.handleEnd(Http1xServerRequest.java:535)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:73)
	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:189)
	at io.vertx.core.http.impl.Http1xServerConnection.onEnd(Http1xServerConnection.java:200)
	at io.vertx.core.http.impl.Http1xServerConnection.onContent(Http1xServerConnection.java:187)
	at io.vertx.core.http.impl.Http1xServerConnection.handleOther(Http1xServerConnection.java:156)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:144)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:151)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:144)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.websocketx.extensions.WebSocketServerExtensionHandler.channelRead(WebSocketServerExtensionHandler.java:101)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by slinkydeveloper** [comment]
> Warning: Failed to download action 'https://api.github.com/repos/actions/setup-go/tarball/37335c7bb261b353407cff977110895fa0b4f7d8'. Error: The operation was canceled.
Warning: Back off 14.354 seconds before retry.
Warning: Failed to download action 'https://api.github.com/repos/actions/setup-go/tarball/37335c7bb261b353407cff977110895fa0b4f7d8'. Error: The operation was canceled.
Warning: Back off 16.144 seconds before retry.

Flaky

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/IllegalResponseException.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 public class IllegalResponseException extends Exception {
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/IllegalResponseException.java:18]: Why overriding `Exception`? Can you override `RuntimeException`?

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/IllegalResponseException.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
...
 
-  public IllegalResponseException(String s) {
-    super(s);
+  public IllegalResponseException(final String s, final Exception ex) {
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/IllegalResponseException.java:20]: When you get a parameter to use as exception cause, take always `Throwable`

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpSinkResponseHandler.java [hunk 3]
```diff
@@ -66,35 +63,28 @@ public HttpSinkResponseHandler(
...
+    } catch (final Exception ex) {
       if (response.body() != null && response.body().length() > 0) {
+        // When the sink returns a malformed event we return a failed future to avoid committing the message to Kafka.
         return Future.failedFuture(
-          new IllegalResponseException("Unable to decode response: unknown encoding and non empty response")
+          new IllegalResponseException("Unable to decode response: unknown encoding and non empty response", ex)
         );
       }
 
       // Response is non-event, discard it
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpSinkResponseHandler.java:86]: Can we have a debug log here that states: "response is not recognized as event, discarding it"

**Review by slinkydeveloper — COMMENTED** [review]
Nits

**Pull request merged**